common/lib/common-utils/src/test/mocha/idleTaskScheduler.spec.ts

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { scheduleIdleTask } from "../../idleTaskScheduler";
+
+describe("Idle task scheduler", () => {
+    function someTask(x: number): void {
+        if (x < 3) {
+            throw new Error("Something awful happened");
+        }
+    }
+
+    it("Should schedule and run a synchronous task during idle time", async () => {
+        let success = false;
+        await scheduleIdleTask(() => {
+            someTask(5);
+        }, 1000)
+            .then(() => {
+                success = true;
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+        assert(success);
+    });
+
+    it("Should fall back to setTimeout when idle Task API is not available", async () => {
```

Comment:
This test does not use scheduleIdleTask, so not clear what exactly in tests.

Reply:
Take a look at https://github.com/microsoft/FluidFramework/pull/12824 - we can mimic various behaviors (presence or lack of global API) and thus test various things (and the fact that we use timers when API is not present, and that we do not use timers when API is present).
Even better would be for test not to actually need to wait idle any time - we can fake time - see usage of useFakeTimers() and similar API in our repo